### PR TITLE
test(rebalance): improve testing coverage of rebalancing

### LINKED ISSUE #8: Better rebalancing/reconstitution test coverage
DOD: Cover this:

- [ ] Changing weights w/o changing token list of a single pair of tokens (1)
- [ ] Changing token list w/o changing weights of a single pair of tokens (2) 
- [ ] (1) with >2 tokens
- [ ] (2) with >2 tokens
- [ ] Changing both weights and tokens

### PR DESCRIPTION
**Pull request by pysel**
Closes: #8 

Checks:

1. Test for >2 tokens rebalance per invokation
2. Test for basic reconstitution between two tokens
3. Test for >2 tokens reconstitution
4. Test both rebalancing and reconstitution during a single invokation

### CODE REVIEW

File: src/TtcVault.sol [hunk 1]
```diff
@@ -371,7 +375,7 @@ contract TtcVault is ITtcVault, ReentrancyGuard {
      */
     function checkTokenList(Token[10] memory _tokens) internal view returns (bool) {
         // Make sure the first token is always rETH
-        if (_tokens[0].tokenAddress != address(i_rEthToken) || _tokens[0].weight != 50) {
+        if (_tokens[0].tokenAddress != address(i_rEthToken) || _tokens[0].weight < 50) {
```
Comment (pysel) [inline src/TtcVault.sol]: note: I think this makes more sense

File: src/TtcVault.sol [hunk 2]
```diff
@@ -508,49 +512,38 @@ contract TtcVault is ITtcVault, ReentrancyGuard {
...
+
+        for (uint8 i; i < 3; i++) {
+            pool = getPoolWithFee(tokenAddress, wEthAddress, feeTiers[i]);
+            if (pool == address(0)) {
+                continue;
+            } else {
+                _pool = IUniswapV3PoolState(pool);
+                break;
+            }
+        }
```
Comment (pysel) [inline src/TtcVault.sol:532]: this is a refactored stuff that was before, looks better imo

File: test/TtcTestContext.sol [hunk 3]
```diff
@@ -87,11 +91,18 @@ contract TtcTestContext is Test {
                 SHIB_ADDRESS
             )
         );
-        // TONCOIN Token
+        // // TONCOIN Token
```
Comment (pysel) [inline test/TtcTestContext.sol:94]: note: toncoin/eth pools have a very low liquidity -> this is why I was getting wrong results from ton/eth price calculation functions

File: src/TtcVault.sol (hunk 1 above)
Comment (ShivaanshK) [reply src/TtcVault.sol]: We stipulate a constant allocation of ETH, so I think an exact check is better.

File: src/TtcVault.sol (hunk 2 above)
Comment (ShivaanshK) [reply src/TtcVault.sol:532]: Agreed

File: test/TtcTestContext.sol (hunk 3 above)
Comment (ShivaanshK) [reply test/TtcTestContext.sol:94]: Ok.

File: src/TtcVault.sol (hunk 1 above)
Comment (pysel) [reply src/TtcVault.sol]: fair, addressed
merging now

**Pull request merged**